cocotb/handle.py

```diff
@@ -442,12 +476,6 @@ def __ne__(self, other):
             return SimHandleBase.__ne__(self, other)
         return self.value != other
 
-    # We want to maintain compatibility with python 2.5 so we can't use @property with a setter
-    value = property(fget=lambda self: self._getvalue(),
-                     fset=lambda self, v: self._setcachedvalue(v),
-                     fdel=None,
-                     doc="A reference to the value")
```

Comment:
Your metaclass above feels like crazy overkill - One simple option would be:
```python
@property
def value(self):
    return self._getvalue()
@value.setter
def value(self, v):
    self._setcachedvalue(v)
```

If you're trying to optimize this further, then there are still tidier ways to do it - I'll put up an alternate patch for this path

Reply:
See gh-1287 for a less complex solution

Reply:
Removed the commit in favor of your PR.